tools/buildFactory.ts

```diff
@@ -174,73 +202,166 @@ export async function gulpUmdMinFactory(input: string, output: string) {
+      : '';
+    if (previousHash === hashValue) return;
+    // generate index file
+    const project = gulpTs.createProject('tsconfig.json', {
+      target: 'ESNext',
+      declaration: true,
+      getCustomTransformers: (program) => {
+        if (program === undefined) throw new Error('program is undefined');
+        return exportTransformer(program, {
+          indexOutputPath: `${indexFileDir}/transformed-index.json`,
+          indexFile: input.indexFile,
+          builds: input.isolateBuildConfig.builds
+            .map((build) => ({ name: build.name, sourceRoots: [build.sourceRoot] }))
+        });
+      }
+    });
+    await new Promise((resolve, reject) => {
+      gulp.src(input.globs, input.opts)
+        .pipe(project())
+        .on('error', reject)
+        .on('end', resolve)
+        .on('finish', resolve);
+    });
+    // save hash
+    fs.mkdirSync(indexFileDir, { recursive: true });
+    fs.writeFileSync(`${indexFileDir}/hash`, hashValue);
   }
-  build.displayName = `${taskPrefix}-build`;
-
-  function filterDependencies() {
-    return gulp.src(`${output}/**/*`, { read: false, allowEmpty: true, base: '.' })
-      .pipe(filter([
-        '**/*.ts',
-        '**/*.js',
-        '**/*.js.map',
-        ...dependencySolver.getIncludedDependencies().map((dep) => `!${dep}`),
-        ...dependencySolver.getIncludedDependencies().map((dep) => `!${dep}.map`)
-      ]))
+
+  generateIndex.displayName = `${format}-generate-index`;
+
+  function cleanBuild() {
+    return gulp.src(output, { read: false, allowEmpty: true })
       .pipe(clean());
   }
-  filterDependencies.displayName = `${taskPrefix}-filter-dependencies`;
 
-  async function cleanEmptyDirs() {
-    await deleteEmpty(process.cwd() + '/' + output);
+  cleanBuild.displayName = `${format}-clean`;
+
+  let originalIndexFile: string|null = null;
+
+  async function replaceIndex() {
+    originalIndexFile = fs.readFileSync(input.indexFile, 'utf8');
+    const transformedIndexFilePaths = `${indexFileDir}/transformed-index.json`;
+    if (!fs.existsSync(transformedIndexFilePaths)) throw new Error('index file does not exist');
+    const transformedIndexFiles =
+      JSON.parse(fs.readFileSync(transformedIndexFilePaths, 'utf8')) as { [key: string]: string };
+    const indexFileContent = transformedIndexFiles[input.buildName];
+    fs.writeFileSync(input.indexFile, indexFileContent);
   }
-  cleanEmptyDirs.displayName = `${taskPrefix}-clean-empty-dirs`;
 
-  return gulp.series(build, filterDependencies, cleanEmptyDirs);
+  replaceIndex.displayName = `${format}-replace-index`;
+
+  async function build() {
+    macros.clear();
+    const rollupBundle = await rollup.rollup({
+      input: input.indexFile,
+      plugins: [
+        rollupPluginTs({
+          browserslist: false,
+          tsconfig: {
+            ...tsConfig.compilerOptions as object,
+            ...overrideSettings
+          },
+          transformers: (options) => {
```

Comment:
Is this config work?

I didn't see the `transformers` parma in `rollupOption`.